### PR TITLE
Add match priority master

### PR DESCRIPTION
**Pull request by rchande**
https://github.com/dotnet/roslyn/pull/11874 against master.
cc @dotnet/roslyn-ide @CyrusNajmabadi

### CODE REVIEW

**Comment by Pilchie** [comment]
Why do we need this?  Won't `stabilization` be merged back for us?

File: src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs [hunk 1]
```diff
@@ -326,17 +408,59 @@ private static StringComparison GetComparision(bool isCaseSensitive)
         }
 
         /// <summary>
+        /// Returns true if the enter key that was typed should also be sent through to the editor
+        /// after committing the provided completion item.
+        /// </summary>
+        public virtual bool SendEnterThroughToEditor(CompletionItem item, string textTypedSoFar, OptionSet options)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs]: Why you make this change :(

File: src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs [hunk 2]
```diff
@@ -313,11 +326,80 @@ protected bool IsArgumentName(CompletionItem item)
...
         }
 
-        private static bool TextTypedSoFarMatchesItem(CompletionItem item, char ch, string textTypedSoFar)
+        /// <summary>
+        /// Returns true if the character is one that can commit the specified completion item. A
+        /// character will be checked to see if it should filter an item.  If not, it will be checked
+        /// to see if it should commit that item.  If it does neither, then completion will be
+        /// dismissed.
+        /// </summary>
+        public virtual bool IsCommitCharacter(CompletionItem item, char ch, string textTypedSoFar, string textTypedWithChar = null)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs]: Why you make this change :(

File: src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs (hunk 1 above)
Comment (rchande) [inline src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs]: That's super weird, all I did was cherry-pick my changes onto master.

**Comment by rchande** [comment]
@CyrusNajmabadi @Pilchie Yeah, given I have no idea what happened when I cherry-picked my changes (something weird), I"ll just keep the one against stabilization and let it flow whenever.

**Pull request closed without merging**